src/libraries/System.Linq.Expressions/src/System/Dynamic/DynamicObject.cs

```diff
@@ -145,7 +146,7 @@ public virtual bool TryInvoke(InvokeBinder binder, object[] args, out object res
         /// <param name="result">The result of the operation.</param>
         /// <returns>true if the operation is complete, false if the call site should determine behavior.</returns>
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1007:UseGenericsWhereAppropriate")]
-        public virtual bool TryBinaryOperation(BinaryOperationBinder binder, object arg, out object result)
+        public virtual bool TryBinaryOperation(BinaryOperationBinder binder, object arg, [NotNullWhen(true)] out object? result)
```

Comment:
>[NotNullWhen(true)] [](start = 89, length = 19)

Can an implementation return `null` for the result of the binary operation?

Reply:
i found only one implementation which is not returning null for result of the binary operation